encodings/runend-bool/src/array.rs

```diff
@@ -65,44 +76,42 @@ impl RunEndBoolArray {
             children.push(a)
         }
 
-        Self::try_from_parts(
-            DType::Bool(Nullability::NonNullable),
-            length,
-            metadata,
-            children.into(),
-            StatsSet::new(),
-        )
+        Self::try_from_parts(dtype, length, metadata, children.into(), StatsSet::new())
     }
 
-    pub fn find_physical_index(&self, index: usize) -> VortexResult<usize> {
+    pub(crate) fn find_physical_index(&self, index: usize) -> VortexResult<usize> {
         search_sorted(&self.ends(), index + self.offset(), SearchSortedSide::Right)
             .map(|s| s.to_ends_index(self.ends().len()))
     }
 
-    pub fn validity(&self) -> Validity {
-        self.metadata().validity.to_validity(|| {
-            self.as_ref()
-                .child(2, &Validity::DTYPE, self.len())
-                .vortex_expect("RunEndBoolArray: validity child")
-        })
-    }
-
     #[inline]
-    pub fn offset(&self) -> usize {
+    pub(crate) fn offset(&self) -> usize {
```

Comment:
these likely should be public so people can use them but we can always loosen it